li-apache-kafka-clients/src/test/java/com/linkedin/kafka/clients/largemessage/MessageAssemblerTest.java

```diff
@@ -60,6 +88,24 @@ public void testTreatBadSegmentAsPayload() {
     Assert.assertEquals(assembleResult.messageEndingOffset(), 0);
   }
 
+  // TODO: discuss with navina about if we should throw exception when treating bad segments.
```

Comment:
I am not sure what your question is here. Why does the validation of `LargeMessageSegment` have anything to do with serialization format of the large message headers?